editor/editor_node.cpp

```diff
@@ -2000,6 +2006,28 @@ void EditorNode::_mark_unsaved_scenes() {
 	scene_tabs->update_scene_tabs();
 }
 
+bool EditorNode::_is_scene_unsaved(int p_idx) {
+	const Node *scene = editor_data.get_edited_scene_root(p_idx);
+	if (!scene) {
+		return false;
+	}
+
+	if (EditorUndoRedoManager::get_singleton()->is_history_unsaved(editor_data.get_scene_history_id(p_idx))) {
+		return true;
+	}
+
+	const String &scene_path = scene->get_scene_file_path();
+	if (!scene_path.is_empty()) {
+		// Check if scene has unsaved changes in built-in resources.
+		for (int j = 0; j < editor_data.get_editor_plugin_count(); j++) {
+			if (!editor_data.get_editor_plugin(j)->get_unsaved_status(scene_path).is_empty()) {
```

Comment:
```suggestion
		for (int i = 0; i < editor_data.get_editor_plugin_count(); i++) {
			if (!editor_data.get_editor_plugin(i)->get_unsaved_status(scene_path).is_empty()) {
```
Unless `j` was chosen because of `p_idx`

Reply:
j was choosen because of outer loop.